packages/webapp/src/components/Animals/AddAnimalsForm/AnimalFormHeaderItem/index.tsx

```diff
@@ -0,0 +1,163 @@
+/*
+ *  Copyright 2024 LiteFarm.org
+ *  This file is part of LiteFarm.
+ *
+ *  LiteFarm is free software: you can redistribute it and/or modify
+ *  it under the terms of the GNU General Public License as published by
+ *  the Free Software Foundation, either version 3 of the License, or
+ *  (at your option) any later version.
+ *
+ *  LiteFarm is distributed in the hope that it will be useful,
+ *  but WITHOUT ANY WARRANTY; without even the implied warranty of
+ *  MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+ *  GNU General Public License for more details, see <https://www.gnu.org/licenses/>.
+ */
+
+import { useState } from 'react';
+import clsx from 'clsx';
+import { useTheme } from '@mui/styles';
+import { useMediaQuery } from '@mui/material';
+import { useTranslation } from 'react-i18next';
+import styles from './styles.module.scss';
+import { IconLink, Main } from '../../../Typography';
+import Icon from '../../../Icons';
+import TextButton from '../../../Form/Button/TextButton';
+import { ReactComponent as TrashIcon } from '../../../../assets/images/animals/trash_icon_new.svg';
+
+type BaseFormHeader = {
+  type: string;
+  breed?: string;
+  iconKey: string;
+  number: number;
+  totalCount: number;
+  showRemove?: boolean;
+  onRemove: () => void;
+  isExpanded: boolean;
+};
+
+type AnimalFormHeader = BaseFormHeader & {
+  sex: string;
+  count?: never;
+};
+
+type BatchFormHeader = BaseFormHeader & {
+  count: number;
+  sex?: never;
+};
+
+type AnimalFormHeaderItemProps =
+  | (AnimalFormHeader & { isBatch: false })
+  | (BatchFormHeader & { isBatch: true });
```

Comment:
I'm just wondering and maybe I'm missing something, but is this better than including isBatch in AnimalFormHeader and BatchFormHeader??